src/main/java/io/aiven/connect/jdbc/sink/BufferedRecords.java

```diff
@@ -169,25 +185,40 @@ public List<SinkRecord> flush() throws SQLException {
                 successNoInfo = true;
                 continue;
             }
-            totalUpdateCount += updateCount;
+            totalSuccessfulExecutionCount += updateCount;
+        }
+        for (final int deleteCount : executeDeleteBatch()) {
+            if (deleteCount == Statement.SUCCESS_NO_INFO) {
+                successNoInfo = true;
+                continue;
+            }
+            totalSuccessfulExecutionCount += deleteCount;
         }
         log.debug("Done executing batch.");
-        if (totalUpdateCount != records.size() && !successNoInfo) {
+        verifySuccessfulExecutions(totalSuccessfulExecutionCount, successNoInfo);
```

Comment:
Should we invoke this once per executed batch (i.e., once for regular records and once for tombstone records)? Otherwise, the `successNoInfo` field gets reused across batches and may produce inaccurate results.